File: src/main/java/z3roco01/molecular/Molecular.java
package z3roco01.molecular;

import net.fabricmc.api.ModInitializer;
import net.minecraft.resources.Identifier;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Molecular implements ModInitializer {
    public static final String MOD_ID = "molecular";
    public static final Logger LOGGER = LoggerFactory.getLogger(MOD_ID);

    @Override
    public void onInitialize() {
        LOGGER.info("#molecules");
    }

    /**
     * Adapts to the {@link Identifier} changes introduced in 1.21.
     */
    public static Identifier id(String namespace, String path) {
        //? if <1.21 {
        /*return new Identifier(namespace, path);
        *///?} else
        return Identifier.fromNamespaceAndPath(namespace, path);
    }
}

File: src/main/kotlin/molecular/Molecular.kt
package molecular

import molecular.item.ElementItem
import molecular.util.ResourceUtil
import net.fabricmc.api.ModInitializer
import net.minecraft.registry.Registries
import net.minecraft.registry.Registry
import net.minecraft.util.Identifier
import org.slf4j.LoggerFactory

object Molecular : ModInitializer {
	const val MOD_ID = "molecular"
    val logger = LoggerFactory.getLogger(MOD_ID)

	override fun onInitialize() {
		var files = ResourceUtil.getFilesInDir("/data/molecular/elements")
		val americium = ResourceUtil.fileToElement(files[0])
		Registry.register(Registries.ITEM, Identifier.of(MOD_ID, "americium"), ElementItem(americium))

		logger.info("common init done ! :D")
	}
}